### ISSUE TITLE
Vue 2.2 update

### ISSUE THREAD

**Issue by mp4096** [tum-rt/web-quiz#29]
Upgrading `vue` und `vue-template-compiler` to `2.2.1` results in build fail:

```
ERROR in ./~/vue-loader/lib/template-compiler.js?{"id":"data-v-05e1d115"}!./~/vue-loader/lib/selector.js?type=template&index=0!./app/components/Snippet.vue

  Error compiling template:
  
  <div>
      <div v-if="snippet.text" class="form-group mt-10">
          <label class="form-label"><fl-text :value="snippet.text"></label>
      </div>
      <h4 v-if="snippet.header1">
          <fl-text :value="snippet.header1">
      </h4>
      <h5 v-if="snippet.header2">
          <fl-text :value="snippet.header2">
      </h5>
      <h6 v-if="snippet.header3">
          <fl-text :value="snippet.header3">
      </h6>
      <div v-if="snippet === 'separator'" class="divider"></div>
      <img v-if="snippet.image" :src="snippet.image" class="img-responsive" />
      <div v-if="snippet.hint" class="hint">
          <button v-show="!showHint" class="btn btn-block" @click="triggerHint">{{ $t("show_hint") }}</button>
          <span v-show="showHint" class="toast">
              <fl-text :value="snippet.hint">
          </span>
      </div>
      <div v-if="snippet.task_text" class="input-group">
          <span v-if="snippet.task_text.before" class="input-group-addon">
              <fl-text :value="snippet.task_text.before">
          </span>
          <input type="text" class="form-input" v-model="taskText">
          <span v-if="snippet.task_text.after" class="input-group-addon">
              <fl-text :value="snippet.task_text.after">
          </span>
      </div>
      <div v-if="snippet.task_number" class="input-group">
          <span v-if="snippet.task_number.before" class="input-group-addon">
              <fl-text :value="snippet.task_number.before">
          </span>
          <input type="number" class="form-input" v-model="taskNumber">
          <span v-if="snippet.task_number.after" class="input-group-addon">
              <fl-text :value="snippet.task_number.after">
          </span>
      </div>
      <template v-if="snippet.task_mc_multiple_correct">
          <div v-for="(answer, answerIndex) in snippet.task_mc_multiple_correct.answers"
              class="form-group">
              <label class="form-checkbox quiz-answer-box" :class="{'quiz-answer-box-highlight': taskMCMultiple.indexOf(answerIndex) >= 0}">
                  <i class="icon-check-square-o" v-if="taskMCMultiple.indexOf(answerIndex) >= 0"></i>
                  <i class="icon-square-o" v-else></i>
                  <input type="checkbox" v-model="taskMCMultiple" :value="answerIndex">
                  <fl-text :value="answer.value">
              </label>
          </div>
      </template>
      <template v-if="snippet.task_mc_one_correct">
          <div v-for="(answer, answerIndex) in snippet.task_mc_one_correct.answers"
              class="form-group">
              <label class="form-radio quiz-answer-box" :class="{'quiz-answer-box-highlight': taskMCOne === answerIndex}">
                  <input type="radio" :name="index" :value="answerIndex" v-model="taskMCOne">
                  <i class="icon-dot-circle-o" v-if="taskMCOne == answerIndex"></i>
                  <i class="icon-circle-o" v-else></i>
                  <fl-text :value="answer.value">
              </label>
          </div>
      </template>
  </div>
  
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.
  - tag <fl-text> has no matching end tag.

 @ ./app/components/Snippet.vue 5:2-178
 @ ./~/vue-ts-loader!./~/vue-loader/lib/selector.js?type=script&index=0!./app/components/Question.vue
 @ ./app/components/Question.vue
 @ ./~/vue-ts-loader!./~/vue-loader/lib/selector.js?type=script&index=0!./app/components/App.vue
 @ ./app/components/App.vue
 @ ./app/app.ts
```

**Comment by mp4096** [comment]
See https://github.com/vuejs/vue/releases/tag/v2.2.0:

> When using v-for with a component, a key is now required. You will likely see a bunch of "soft warnings" when you upgrade, but this does not affect the current behavior of your app.

> The template parser will now raise errors for tags that are missing matching closing tags. Although a tag without matching closing tag is vaild in HTML5, it is most of the time an unintentional mistake, and can lead to subtle bugs. This check does not work for in-DOM templates because the HTML string is already normalized by the browser before being passed to Vue's parser.

**Comment by mp4096** [comment]
I think I'll try to fix this one.

**Comment by syxolk** [comment]
Yes, just do ` <fl-text ... />` instead of ` <fl-text ...>`

**Comment by mp4096** [comment]
Ok, this wasn't the only breaking change. I also had to reconfigure Vue import, since it couldn't load `vueI18n` (_`Vue.use` is not a function_). 

However, I couldn't follow [these guidelines](https://vuejs.org/v2/guide/typescript.html) exactly. Adding

```json
"lib": [
    "dom",
    "es5",
    "es2015.Promise"
]
```

to `tsconfig.json` leads to compilation errors.

**Comment by mp4096** [comment]
See 7aa73c8ebdfe2ccc98671f930c68cb7ef3a17f1b

**Comment by mp4096** [comment]
Awww snap: https://travis-ci.org/tum-rt/web-quiz/jobs/208606605

**Comment by mp4096** [comment]
Ok, I guess I did as much as I could. @syxolk could you take over from here please?

**Comment by syxolk** [comment]
So only the Karma tests fail?

**Comment by syxolk** [comment]
Probably fixed with 921673c4ed1ba1c5ee724525d048c2084caf20aa

**Comment by mp4096** [comment]
Ok, I've cleaned up `tsconfig.json`. I decided to add as little as possible for the time being. I'm not sure how adding all these `es2015` options affects compatibility with older browsers.

Anyway, we should be ready to refactor in the future if Vue deprecates something.
